### PR TITLE
Fix 30

### LINKED ISSUE #30: `VolBlockIter::nth` implementaiton always skip one too many element
current implementation is
```rust
  fn nth(&mut self, n: usize) -> Option<Self::Item> {
    if n < self.count {
      self.count -= n;
      self.base = unsafe { self.base.add(1 + n) };
      Some(self.base)
    } else {
      self.count = 0;
      None
    }
  }

```
However, to reflect the `next` implementation, it should be:
```rust
  fn nth(&mut self, n: usize) -> Option<Self::Item> {
    if n < self.count {
      self.count -= n;
      self.base = unsafe { self.base.add(n) };
      self.next()
    } else {
      self.count = 0;
      None
    }
  }

```

A good test case is `iter.nth(0)`, it should return the same value as if `iter.next()` was called, but with the current implementation, it returns one after the next.

Edit: also true of `nth_back`

### PR DESCRIPTION
**Pull request by Lokathor**
Fixes #30

### CODE REVIEW

**Comment by Lokathor** [comment]
@nicopap does this look right? I've got it so that all code paths should go through `nth` now, as the single source of truth.

File: src/volseries.rs [hunk 1]
```diff
@@ -296,8 +288,10 @@ impl<T, R, W, const S: usize> core::iter::DoubleEndedIterator
   #[inline]
   fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
     if n < self.count {
+      let out =
+        Some(unsafe { self.base.cast::<[u8; S]>().add(1 + n).cast::<T>() });
```
Comment (nicopap) [inline src/volseries.rs]: Shouldn't this be like the `VolBlockIter::nth_back` implementation?
```suggestion
        Some(unsafe { self.base.cast::<[u8; S]>().add(self.count - (n + 1)).cast::<T>() });
```

File: src/volseries.rs [hunk 2]
```diff
@@ -296,8 +288,10 @@ impl<T, R, W, const S: usize> core::iter::DoubleEndedIterator
   #[inline]
   fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
```
Comment (nicopap) [inline src/volseries.rs:281]: Also `next_back` should be updated to use `nth_back`, similarly to `VolBlockIter`

**Review by nicopap — CHANGES_REQUESTED** [review]
Good idea to use `nth` as a single source of truth. I think the `VolBlockIter` should work now, but I caught some errors in `VolSeriesIter`.

File: src/volseries.rs (hunk 2 above)
Comment (Lokathor) [reply src/volseries.rs:289]: Ah, yeah, I copied from one into the other to make sure the expression patterns matched and made them match *too* much.

File: src/volseries.rs (hunk 1 above)
Comment (Lokathor) [reply src/volseries.rs]: Ah, yeah, I copied from one into the other to make sure the expression patterns matched and made them match too much.

**Review by nicopap — APPROVED** [review]
LGTM

**Comment by Lokathor** [comment]
released 1.2.2

yanked the 0.4 versions and all previous 1.0 versions.

**Pull request merged**